subprojects/core/src/main/java/org/gradle/process/internal/ExecHandleRunner.java

```diff
@@ -57,14 +65,116 @@ public void abortProcess() {
             aborted = true;
             if (process != null) {
                 streamsHandler.disconnect();
-                LOGGER.debug("Abort requested. Destroying process: {}.", execHandle.getDisplayName());
-                process.destroy();
+                LOGGER.info("Abort requested. Destroying process: {}.", execHandle.getDisplayName());
+                if (!OperatingSystem.current().isWindows()) {
+                    destroyProcessTree();
+                } else {
+                    Long pid = getProcessPid();
+                    if (pid == null) {
+                        LOGGER.info("Aborting {}", execHandle.getDisplayName());
+                        process.destroy();
+                    } else {
+                        // taskkill requires pid
+                        destroyWindowsProcessTree(pid);
+                    }
+                }
             }
         } finally {
             lock.unlock();
         }
     }
 
+    private @Nullable Long getProcessPid() {
+        long pid;
```

Comment:
```suggestion
        Long pid;
```

Reply:
Why's that?

Reply:
`java.lang.Process#pid` returns `long`, so I do not see why `Long` might be needed here.

Reply:
The method returns `Long` anyway, and the reflection call casts to `Long`.

Reply:
OptionalInt?

Reply:
Good idea, it might better convey the intent.